### PR TITLE
Bug 1883595: Fix chart merging and show chart provider in catalog

### PR DESCRIPTION
**Pull request by rohitkrai03**
**Depends on**: https://github.com/openshift/console/pull/6753

**Fixes**: https://issues.redhat.com/browse/ODC-4898

**Analysis / Root cause**: The helm charts were getting merged in backend for every chart repo. So, if there was a chart with same name coming from two different chart repos it would get merged and only come as one.

**Solution Description**: https://github.com/openshift/console/pull/6753 updates the merging logic and adds chart repo name to the key for each chart entry in index.yaml. UI parses this chart repo name from the key to display chart repo name as provider for the catalog item. Also updated the version merging logic.

This PR also changes the way we handle chart versions on install/upgrade form.
- Added extra information of chart repo name `(Provided by Chart Repo Name)`
- For Install form we get the chart repo name from the catalog as URL param and we use that to render the version dropdown items. 
- In case of upgrade form we have no information about the chart repo so we show all the chart versions from all the repos with repo name added to the dropdown item.

**Screen shots / Gifs for design review**: 

@openshift/team-devconsole-ux @rachael-phillips @parvathyvr @serenamarie125 

Peek 2020-09-29 17-01

Peek 2020-09-28 21-37

**Unit test coverage report**: 

Screenshot from 2020-09-29 18-15-54

image

**Test setup:**

Build master with https://github.com/openshift/console/pull/6753

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx:113]: Uncomment this or remove it if not needed

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx]: Done.

**Comment by divyanshiGupta** [comment]
@rohitkrai03 Tried it out locally, works fine for me. Will add LGTM once we have UX approval.

**Review by parvathyvr — APPROVED** [review]
lgtm!

**Comment by rachael-phillips** [comment]
lgtm!

**Pull request merged**
